packages/core/saved-objects/core-saved-objects-server-internal/src/routes/utils.ts

```diff
@@ -157,7 +157,9 @@ export interface BulkGetItem {
 export function isKibanaRequest({ headers }: KibanaRequest) {
   // The presence of these two request headers gives us a good indication that this is a first-party request from the Kibana client.
   // We can't be 100% certain, but this is a reasonable attempt.
-  return headers && headers['kbn-version'] && headers.referer;
+  return (
+    headers && headers['kbn-version'] && headers.referer && headers['x-elastic-internal-origin']
```

Comment:
ATM, there are 4: core usage stats client,  cases plugin, rule registry, and spaces usage stats client. I've added the header to core's implementation.
 
Are the concerns Joe raised still val a valid reason to have 4 implementations of `isKibanaRequest` or can we add the extra header to that check? 

A while back, Joe raised [concerns](https://github.com/elastic/kibana/issues/85869) about adding a header to all Kibana requests:
1. other integrations might attempt to use this header
2. adding a new header might present problems for users with reverse proxies that may strip headers not in an allow-list; this would cause false negative for first-party request detection

If these are still valid, then having different implementations is warrented. @response-ops @kibana-security would a single method fullfil your needs or are concerns around false negatives be an issue for you?

Reply:
IMO (1) and (2) should be mitigated because we will not be restricting access to internal endpoints by default so everything should continue working as-is for on-prem.